### PR TITLE
@FormBuilder

### PR DESCRIPTION
**Pull request by RomanPodymov**
Hello.
Thank you for Eureka.
Add `@FormBuilder`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -228,6 +228,24 @@ form +++ {
 }
 ```
 
+#### @FormBuilder
+```swift
+@FormBuilder
+var form: Form {
+    nil as Section?
```
Comment (mats-claassen) [inline README.md:235]: Why this?

File: README.md [hunk 2]
```diff
@@ -228,6 +228,24 @@ form +++ {
...
+    Section("Section A") { section in
+        section.tag = "Section_A"
+    }
+    if true {
+        Section("Section B") { section in
+            section.tag = "Section_B"
+        }
+    }
+    nil as TextRow?
```
Comment (mats-claassen) [inline README.md:244]: What does this do?

Comment (RomanPodymov) [reply README.md:244]: Hello @mats-claassen 
Fixed.

Comment (mats-claassen) [reply README.md]: Hi, so my question is why do we need to put a nil there?

Comment (RomanPodymov) [reply README.md]: What about `if something { nil }` ?

Comment (RomanPodymov) [reply README.md]: Hello @mats-claassen 
OK, I can remove
```
public static func buildOptional(_ components: [RowsProvider]?) -> [BaseRow] {
    components?.flatMap { $0.rows } ?? []
}
```
and
```
public static func buildOptional(_ components: [SectionsProvider]?) -> [Section] {
    components?.flatMap { $0.sections } ?? []
}
```

Comment (RomanPodymov) [reply README.md]: Or should I leave them and remove `if something { nil }` from the examples?

Comment (mats-claassen) [reply README.md]: I think buildOptional is correct. I just don't think that it is an intuititve example to add `nil` to a form. You could add an optional variable in the example which could be nil but there is no case when you would explicitly put `nil` there, right?

Comment (RomanPodymov) [reply README.md]: Hello @mats-claassen 
Exactly. So, should we remove it from the examples?

Comment (RomanPodymov) [reply README.md]: Or you can add your suggestions to my branch.

Comment (RomanPodymov) [reply README.md]: Hello @mats-claassen 
Can you please tell me what else should I fix in this pull request?

Comment (mats-claassen) [reply README.md]: Hey @RomanPodymov I think if you remove the `if something { nil}` from the examples then this is fine to be merged

Comment (RomanPodymov) [reply README.md]: Hello @mats-claassen 
Done.

**Comment by RomanPodymov** [comment]
Hello @mats-claassen 
What other changes should I make in this pull request?

**Pull request merged**